### PR TITLE
6 react dojo

### PR DESCRIPTION
**Pull request by martonszel**

### CODE REVIEW

File: React-Dojo/react_typescript/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+} from "react-router-dom";
+
+import ClassBased from './pages/ClassBased';
+import Functional from './pages/Functional';
+import MainNavigation from './components/MainNavigation';
+
+const App: React.FC = () => {
+
+  const API_URL = 'https://gateway.marvel.com/v1/public/characters'
+  const API_KEY = '&apikey=7fb70a9709792d94b28b122a317fe87f'
```
Comment (KaralyosBela) [inline React-Dojo/react_typescript/src/App.tsx:15]: Your API key is exposed on github, so others can access and use it. Try to save it in a .env file and use it from there. Also don't forget to add the .env file to gitignore.

**Review by KaralyosBela — APPROVED** [review]

File: React-Dojo/react_full_course_2022/package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "react_full_course_2022",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.16.4",
```
Comment (KaralyosBela) [inline React-Dojo/react_full_course_2022/package.json:6]: I think testing relating packages should be in `devDepencies`.

Comment (martonszel) [reply React-Dojo/react_full_course_2022/package.json:6]: https://www.youtube.com/watch?v=T_x6QmuJdms

create react app did this :D :D

File: React-Dojo/react_typescript/src/pages/Functional.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    } else if (!isLoaded) {
+        return <div className='classBased'><h1>Loading...</h1></div>
+    } else {
+        return (
+            <div className='funcContainer'>
+                <h1>Functional</h1>
+                <div className="searchContainer">
+                    <div className="search">
+                        <input placeholder='Search for characters...' value={searchTerm} onChange={(e) => setSearchTerm(e.target.value)} />
+                        <img src={SearchIcon} alt="search" onClick={() => searchCharacter(searchTerm)} />
```
Comment (szkertesz) [inline React-Dojo/react_typescript/src/pages/Functional.tsx:73]: I suggest to use a `<button>` here to attach the click listener to (the `` is not an interactive element).

File: React-Dojo/react_typescript/src/components/MainNavigation.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { Link } from "react-router-dom";
+import classes from './MainNavigation.module.css'
+
+type Props = {}
+
+const MainNavigation = (props: Props) => {
```
Comment (szkertesz) [inline React-Dojo/react_typescript/src/components/MainNavigation.tsx:6]: Are the `props` required here?

**Review by szkertesz — APPROVED** [review]

File: React-Dojo/react_typescript/src/pages/Functional.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+const Functional: React.FC<Props> = ({ apiUrl, apiKey }) => {
+
+    const [characters, setCharacters] = useState<Character[]>([])
+    const [error, setError] = useState<null | Error>(null);
+    const [isLoaded, setIsLoaded] = useState<boolean>(false);
+    const [searchTerm, setSearchTerm] = useState<string>('')
+
+    const getAllCharacters = async () => {
+
+        try {
```
Comment (szkertesz) [inline React-Dojo/react_typescript/src/pages/Functional.tsx:21]: You don't throw any errors here.

File: React-Dojo/react_typescript/src/App.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+  return (
+    <div className="App">
+      <MainNavigation></MainNavigation>
```
[the remark this answers is not in the document]
Comment (martonszel) [reply React-Dojo/react_typescript/src/App.tsx:19]: corrected

File: React-Dojo/react_typescript/src/components/MainNavigation.tsx (hunk 4 above)
Comment (martonszel) [reply React-Dojo/react_typescript/src/components/MainNavigation.tsx:6]: nope, good point

File: React-Dojo/react_typescript/src/pages/Functional.tsx (hunk 5 above)
Comment (martonszel) [reply React-Dojo/react_typescript/src/pages/Functional.tsx:21]: why should i throw an error here ?

File: React-Dojo/react_full_course_2022/src/App.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+const API_URL = 'http://www.omdbapi.com/?apikey=9fc736b6'
+
+const App = () => {
+
+  const [movies, setMovies] = useState([])
+  const [searchTerm, setSearchTerm] = useState('')
+
+  const searchMovies = async (title) => {
+
+    const response = await fetch(`${API_URL}&s=${title}`)
```
Comment (hegedus-adam) [inline React-Dojo/react_full_course_2022/src/App.js:15]: usually it is necessary to `try - catch` the fetch logic

File: React-Dojo/react_full_course_2022/src/App.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  useEffect(() => {
+    searchMovies('Spiderman')
+  }, [])
+
+  return (
+    <div className="app">
+      <h1>Movieland</h1>
+      <div className="search">
+        <input placeholder='Search for movies...' value={searchTerm} onChange={(e) => setSearchTerm(e.target.value)} />
+        <img src={SearchIcon} alt="search" onClick={() => searchMovies(searchTerm)} />
```
Comment (hegedus-adam) [inline React-Dojo/react_full_course_2022/src/App.js:30]: the onClick event handler could be written in a different and more futureproof way

```
#somewhere in the code before the return
const onImageClickHandler = () => searchMovies(searchTerm);

...
<img ... onClick={onImageClickHandler} />
```

File: React-Dojo/react_meetup_maximilian/src/pages/AllMeetups.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+    useEffect(() => {
+        setIsLoading(true)
+        fetch('https://react-meetup-8f553-default-rtdb.europe-west1.firebasedatabase.app/meetups.json').then((response) => {
+            return response.json()
+        }).then((data) => {
+
+            const meetups = []
+
+            for (const key in data) {
```
Comment (hegedus-adam) [inline React-Dojo/react_meetup_maximilian/src/pages/AllMeetups.js:17]: this works, however a really old approach and there are easier ways to do it

```js
const meetups = Object.keys(data).map(key => ({
  ...data[key],
  id: key,
}));
```

**Review by hegedus-adam — APPROVED** [review]

**Pull request merged**